corehq/tests/test_classcleanup.py

```diff
@@ -0,0 +1,142 @@
+from functools import wraps
+from unittest import TestCase
+
+from nose.plugins import PluginTester
+from testil import eq
+
+from .noseplugins import classcleanup as mod
+
+
+class TestClassCleanupPlugin(PluginTester, TestCase):
+    activate = ''  # Activate option not needed. Plugin is always enabled.
+    plugins = [mod.ClassCleanupPlugin()]
+
+    def setUp(self):
+        pass  # super().setUp() is called by self.run_with_errors(...)
+
+    def makeSuite(self):
+        class Test(TestCase):
+            @classmethod
+            @maybe_error(self)
+            def setUpClass(cls):
+                cls.addClassCleanup(lambda: self.result.append("cleanup"))
+
+            @maybe_error(self)
+            def setUp(self):
+                pass
+
+            @maybe_error(self)
+            def runTest(self):
+                pass
+
+            @maybe_error(self)
+            def tearDown(self):
+                pass
+
+            @classmethod
+            @maybe_error(self)
+            def tearDownClass(cls):
+                pass
+
+        return [Test()]
+
+    def run_with_errors(self, *errors, error_class=Exception):
+        self.errors = errors
+        self.result = []
```

Comment:
Nit: `result` is fairly generic here. It might be more straight-forward if it was like `executedSteps`, so that the assertion would be `eq(self.executedSteps, ['setUpClass', ...])`